### PR TITLE
Detailpage card

### PR DESCRIPTION
**Pull request by salaos**

### CODE REVIEW

**Review by Satttoshi — APPROVED** [review]
Please fix Vercel Deployment, something wrong with your code.

File: components/Challenges.js [hunk 1]
```diff
@@ -2,16 +2,56 @@
...
+          <Card
+            key={challenge.id}
+            challenge={challenge}
+            onClick={() => toggleExpand(challenge.id)}
+          >
+            <h2>{expandedChallenge === challenge.id ? "▼" : "▶︎"}</h2>
             <h2>{challenge.title}</h2>
+
+            {expandedChallenge === challenge.id && (
+              <>
```
[the remark this answers is not in the document]
Comment (Satttoshi) [reply components/Challenges.js:33]: Maybe this is the thing, which makes your deployment not work

**Review by Satttoshi — CHANGES_REQUESTED** [review]
stated above

File: components/ChallengeForm.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
 } from "../components/StyledComponents";
 
 async function persistAddedChallenge(challenge) {
-  const baseUrl = "http://localhost:3000";
+  const baseUrl = process.env.MONGODB_URI;
```
Comment (PReichetanz) [inline components/ChallengeForm.js:13]: I'm not sure what's happening here: why do save a baseURL in your .env file and call it MongoDB_URI? As far as I can see, you don't have a backend so far. Should we speak about this in person?

File: server.js
Comment (PReichetanz) [inline server.js:1]: You don't need to use express, is there a reason why you've done it?

File: components/Challenges.js [hunk 3]
```diff
@@ -2,16 +2,56 @@
...
+                <p>
+                  <h3>Level: {challenge.level} </h3>
+                </p>
+                <p>
+                  <h3>Info: </h3>
+                  {challenge.description}
+                </p>
```
Comment (PReichetanz) [inline components/Challenges.js:40]: I think it's not a good practice to nest an `h3` in a `p` tag. If you need a wrapper around the heading, use a `div` instead.

File: components/Challenges.js [hunk 4]
```diff
@@ -2,16 +2,56 @@
...
+                <p>
+                  <h3>Best Case: </h3>
+                  {challenge.bestcase}
+                </p>
```
Comment (PReichetanz) [inline components/Challenges.js:44]: Either use

```jsx
<div>
    <h3>Best Case: </h3>
    <p>{challenge.bestcase}</p>
</div>
```
or remove the `div` if it is not necessary for styling.

File: components/StyledComponents.js [hunk 5]
```diff
@@ -151,6 +151,44 @@ export const StyledButton = styled.button`
...
+
+    a {
+      color: var(--primary);
+    }
+  }
+
+  a {
+    color: white;
+    font-size: 2rem;
+  }
```
Comment (PReichetanz) [inline components/StyledComponents.js:189]: When do you need an `<a>` tag inside of a button? I think in the case you need something to click on and being redirected, you should rather use a next/link instead.

File: pages/form.js [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
+export default function FormPage({ onAddChallenge }) {
   return (
     <main>
-      <ChallengeForm />
-      <h3>
-        <Link href="..">Back to challenges</Link>
-      </h3>
+      <ChallengeForm onAddChallenge={onAddChallenge} />
+      <Link href="/" aria-label="back">
+        <RoundButton>❮</RoundButton>
```
Comment (PReichetanz) [inline pages/form.js:11]: The `RoundButton` component is an HTML `button` element, which is unnecessary inside of a Link. I suggest to rather use a styled `<span>` to be more semantic. The rest is great!

File: pages/index.js [hunk 7]
```diff
@@ -1,6 +1,18 @@
...
-export default function Challenges() {
-  return <main></main>;
+export default function Challenges({ challenges }) {
+  return (
+    <main>
+      <h1>My Challenges</h1>
+
+      <ChallengeList challenges={challenges} />
+      <Link href="/form" aria-label="add new challenge">
+        <RoundButton>✚</RoundButton>
```
Comment (PReichetanz) [inline pages/index.js:14]: This should not be a button, but rather a span, see my comment above.

File: styles.js [hunk 8]
```diff
@@ -73,12 +73,12 @@ header {
...
-    font-size: 1.4rem;
-  } @media (min-width: 1024px) {
-      font-size: 1.6rem;
-  }}
+  // h3 { font-size: 1.2rem;
+  //   @media (min-width: 768px) {
+  //   font-size: 1.4rem;
+  // } @media (min-width: 1024px) {
+  //     font-size: 1.6rem;
+  // }}
```
Comment (PReichetanz) [inline styles.js:81]: if you don't need it, you can remove it.

**Review by PReichetanz — COMMENTED** [review]
Really nice work! I left some comments on semantic HTML, feel free to contact me 🚀

**Pull request merged**